### PR TITLE
fixed the bug that messageAdapter can not work for primitive types

### PR DESCRIPTION
**Pull request by pxsdirac**
References #28112

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
At least one pull request committer is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

**Review by helena — APPROVED** [review]
LGTM

File: akka-actor-typed/src/main/scala/akka/actor/typed/internal/ActorContextImpl.scala [hunk 1]
```diff
@@ -220,8 +219,9 @@ import org.slf4j.LoggerFactory
   private def internalMessageAdapter[U](messageClass: Class[U], f: U => T): ActorRef[U] = {
     // replace existing adapter for same class, only one per class is supported to avoid unbounded growth
     // in case "same" adapter is added repeatedly
-    _messageAdapters = (messageClass, f.asInstanceOf[Any => T]) ::
-      _messageAdapters.filterNot { case (cls, _) => cls == messageClass }
+    val boxedMessageClass = BoxedType(messageClass).asInstanceOf[Class[U]]
+    _messageAdapters = (boxedMessageClass, f.asInstanceOf[Any => T]) ::
+      _messageAdapters.filterNot { case (cls, _) => cls == boxedMessageClass }
```
Comment (patriknw) [inline akka-actor-typed/src/main/scala/akka/actor/typed/internal/ActorContextImpl.scala]: good, the BoxedType is also used in BehaviorInterceptor for similar reason

**Review by patriknw — APPROVED** [review]
LGTM, thanks for fixing

**Comment by johanandren** [comment]
OK TO TEST

**Comment by helena** [comment]
CI test failed on `akka.actor.typed.NormalActorContextSpec: An ActorContext must canonicalize behaviors`

**Comment by pxsdirac** [comment]
> LGTM, thanks for fixing

@

> CI test failed on `akka.actor.typed.NormalActorContextSpec: An ActorContext must canonicalize behaviors`

I'll follow on that.

**Comment by lightbend-cla-validator** [comment]
At least one pull request committer is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

**Comment by pxsdirac** [comment]
is there anything different from my local env testing and ci testing. It is passed when I do testing in my laptop.

**Comment by raboof** [comment]
That latest test failure is #28147 and has nothing to do with this PR, sorry!

**Comment by raboof** [comment]
PLS BUILD

**Comment by pxsdirac** [comment]
I did sign the CLA

**Comment by lightbend-cla-validator** [comment]
At least one pull request committer is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

**Comment by johanandren** [comment]
I think the problem is that the commits are in a different name/email than your GitHub account (
xusheng). You should be able to align the commit authors with git somehow, but I don't remember exactly how.

**Comment by pxsdirac** [comment]
replace this PR with #28188

**Pull request closed without merging**